src/coreclr/jit/emitxarch.h

```diff
@@ -720,7 +729,7 @@ inline emitter::opSize emitEncodeScale(size_t scale)
 {
     assert(scale == 1 || scale == 2 || scale == 4 || scale == 8);
 
-    return emitSizeEncode[scale - 1];
+    return emitSizeEncode[genLog2((unsigned int)scale)];
```

Comment:
Same as above, I don't think we need the memory lookup here as well.

Just `return static_cast<emitter::opSize>(genLog2(scale))` should be sufficient since `opSize` is `OPSZ1 = 0, ..., OPSZ64 = 6`

Reply:
Done